### ISSUE TITLE
2.2 - 2.5x Speed issue - Pytorch vs. Lua Torch

### ISSUE THREAD

**Issue by george-gssy** [pytorch/pytorch#15794]
Hello,

I've just completed a Pytorch implementation of a Lua code using Torch.
My base programming language is Python, thus I used a lot of optimization keeping in mind
the speed considerations. It contains a lot of torch.tensor initializations and operations.
Finally I ran some tests and I found that the Lua version is 2.2-2,5 faster than the Python version.
Is it possible that the Lua/Torch version is faster in general regarding the tensor operations?

Thank you for your help in advance.

**Comment by albanD** [comment]
Hi,

That would depend a lot on what you do. All the heavy computations are done in cpp so they will be similar.
But the wrapping code might have a large impact in your usecase.

You should create a new post on the [forum](https://discuss.pytorch.org/) with your exact use case so that we can discuss there the possible reasons for the problem.

I'll close this for now as it is not a bug report. We might re-open if after discussion on the forum it appears that there is a bug in pytorch.

**Comment by george-gssy** [comment]
@albanD First, thank you for your time. I will create a new post, as I wanted to do it in pytorch, though I expected some speed loss because of wrapping and I agreed happily up to some extent for Python, but obviously this is far too much to handle it.
Thank you for your help again.
PS: wrapping code, you meant pytorch wrapping code, didn't you?

**Comment by albanD** [comment]
Thanks! Using the forum will reduce the noise on github.

Yes i mean both the python code that runs and the different layers between the python call and the low level c code.
